src/tools/common/sss_tools.c

```diff
@@ -108,18 +112,14 @@ static errno_t sss_tool_confdb_init(TALLOC_CTX *mem_ctx,
         return ret;
     }
 
-    ret = confdb_init(mem_ctx, &confdb, path);
+    ret = confdb_init(mem_ctx, _confdb, path);
```

Comment:
Not directly related to you commit.

`confdb_init()` is assuming its `cdb_ctx` parameter is not `NULL`. It clearly is not here, but could be in other calls. Shouldn't we also check inside this function that it is not `NULL`? Or maybe just move the check to that function?

Reply:
> Shouldn't we also check inside this function that it is not NULL?

this == `sss_tool_confdb_init()`?

But here is the check in place at the beginning (line 97)
Or do I misunderstand?

Reply:
No, sorry. this == `confdb_init()`.

In other words, `sss_tool_confdb_init()` is checking `_confdb != NULL`, but other functions are also calling `confdb_init()`, and those functions may or may not check that condition. It might be better to move the check inside `confdb_init()`.

Reply:
Ok, added one more patch.